src/main/java/seedu/address/model/project/TodoList.java

```diff
@@ -69,7 +69,12 @@ public void deleteTodo(Integer i) {
      */
     public void markAsDone(Integer i) {
         requireNonNull(i);
-        this.todos.get(i).markAsDone();
+        CompletableTodo todo = todos.get(i);
+        todo.markAsDone();
+
+        // Force observable list to update
+        this.todos.remove(todo);
```

Comment:
Do you wanna make this change to EventList too?

Reply:
No need, only once we implement edit event list. (YAGNI principle)
I want to try an immutable design for events, todos, deadlines in another PR